### PR TITLE
ManageBookings PR

### PR DESCRIPTION
**Pull request by wkonit**
### What this PR does (required):
Made the UI for the manage bookings page. Still needs some tweaking.

### Screenshots / Videos (front-end only):
Screenshot 2022-02-22 182002

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")
- Need to be logged in then navigate the the managebookings page via the url (eg: http://localhost:3000/managebookings)
- Need to install dependencies such as: 
  npm install @mui/material @emotion/react @emotion/styled
  npm install @mui/lab
  npm install date-fns --save

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
  I was not able to figure out to set the CurrentBookings and the PastBookings part yet

- List any problems this PR may cause for the project or other active PRs
  Since I messed up with how I cloned the repo, there might have to be a couple of merges to do?

- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: client/src/pages/ManageBookings/CalenderView.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import useStyles from './useStyles'
+import { takeMonth }from './Calender'
+import { Button, Grid, IconButton, Paper } from '@mui/material'
+import {format, isSameMonth, isSameDay, addMonths, subMonths} from 'date-fns'
+import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+
+
+// const minDate = new Date('2020-01-01T00:00:00.000');
+// const maxDate = new Date('2034-01-01T00:00:00.000');
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/CalenderView.tsx:11]: Let's remove this 😄

File: client/src/pages/ManageBookings/CalenderView.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    function dayColor(day:Date){
+      if(!isSameMonth(day, selectedDate)) return '#CCCCCC';
+      if(isSameDay(day, selectedDate)) return '#FFFFFF';
+      return '#000000';
+    }
+
+    function dayBGColor(day:Date){
+      if(!isSameMonth(day, selectedDate)) return '#FFFFFF';
+      if(isSameDay(day, selectedDate)) return '#f14140';
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/CalenderView.tsx:28]: Okay, so you have the right idea but the red background is actually showing days they have bookings available

File: client/src/pages/ManageBookings/CurrentBookings.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import SettingsIcon from '@mui/icons-material/Settings'
+
+import useStyles from './useStyles';
+
+import Avatar from '@mui/material/Avatar';
+
+
+
+
+const CurrentBookings = () => {
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/CurrentBookings.tsx:17]: Lots of whitespace here, could we clean that up?

File: client/src/pages/ManageBookings/CurrentBookings.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        <div>
+            <Card sx={{width: 500, height: 200}}>
+                
+                <CardHeader
+                
+                    action={
+                        <IconButton>
+                            <SettingsIcon/>
+                        </IconButton>
+                    }
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/CurrentBookings.tsx:41]: Same here

File: client/src/pages/ManageBookings/CurrentBookings.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                    titleTypographyProps={{ variant: "subtitle2", pb: 2, sx: {fontWeight: 600, fontFamily: "Arial"}}}
+                    title = "YOUR NEXT BOOKING"
+                    subheaderTypographyProps={{ variant:"h6", sx: {fontWeight: 600, color:"black"}}}
+                    subheader = {dummyBookings.map((dummyBooking) => dummyBooking.date)}
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/CurrentBookings.tsx:45]: Could we not have space between the equals

File: client/src/pages/ManageBookings/ManageBookings.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+const ManageBookings = () => {
+    const classes = useStyles()
+    const data = CurrentBookings()
+    return (
+        <Container>
+            <Grid container> 
+                <Grid columns={1} justifyContent="flex-start" className={classes.GridTop}>
+                    <NextBooking/>
+                    {/* <CurrentBookings/> */}
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/ManageBookings.tsx:17]: Let's remove this

File: client/src/pages/ManageBookings/NextBooking.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+import Avatar from '@mui/material/Avatar';
+
+
+
+
+const NextBooking = () => {
+    const classes = useStyles()
+    return (
+        <div>
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/NextBooking.tsx:20]: So, thinking in React. This probably isn't the best approach, the current bookings and next booking style will be **identical** so we should just make one component but then change the props for each booking but the styling stays the same so we can have this be more maintainable and keep the codebase concise.

File: client/src/pages/ManageBookings/useStyles.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+const useStyles = makeStyles((theme: Theme) => {
+    return{
+        GridTop:{
+            paddingTop: theme.spacing(6)
+        },
+        Avatar:{
+            float: "left"
+        },
+        CalenderWeek:{
```
Comment (moffatethan) [inline client/src/pages/ManageBookings/useStyles.ts:12]: Let's do camelCase for the CSS names

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Comment by wkonit** [comment]
ManageBookings 3-5-2022

New Changes:
- There is test data in `ManageBookings.tsx` to help with mapping the bookings that can be removed once backend integration is done. 
- Currently the gear button switches between `ACCEPTED` AND `DECLINED` when pressed. Can be improved later and icon can be switched out with something better.
- The selected date on the `CalenderView` is currently set to the current day. There is a variable ready to be set for different date. When booking schedules are more clear, the variable can be edited to take more days but it is only set to take one date now. 
- Removed the seperate booking components leaving only one. 
- Removed `useStyles.ts`.

Issues:
- Couldn't figure out why the single digits dates not being centered despite `alignItems:"center"` and` justifyContent:"center"`.
- Currently there is an error where I cannot use the url to access` /managebookings`. Instead I will be constantly redirected to the login page. I bypassed this issue by routing the login page to `/managebookings`. I'm not sure what's causing the issue, hopefully someone can look into it?

Questions:
- I'm not sure how to really edit the scrollbar so it's just the standard windows(?) one. 
- I'm not 100% sure how to resolve the conflicting files for `client/src/App.tsx`.

**Pull request merged**